tests/callbacks_test.cpp

```diff
@@ -180,6 +180,29 @@ TEST_CASE("EarlyStopAtMinLossLambdaCallbackTest", "[CallbacksTest]")
   EarlyStopCallbacksLambdaFunctionTest(smorms3);
 }
 
+TEST_CASE("EarlyStopAtMinLossCustomLambdaTest", "[CallbacksTest]")
+{
+  // Use the 50-dimensional Rosenbrock function.
+  GeneralizedRosenbrockFunction f(50);
+  // Start at some really large point.
+  arma::mat coordinates = f.GetInitialPoint();
+  coordinates.fill(100.0);
+
+  EarlyStopAtMinLoss<arma::mat> cb(
+      [&](const arma::mat& coordinates)
+      {
+        // Terminate if any coordinate has a value less than 10.
+        return arma::abs(coordinates).min();
+      });
```

Comment:
@rcurtin The comparison is done in the `EndEpoch` function inside the  `EarlyStopCallback`. We only need the least coordinates to examin if it is less than `10`

Reply:
Oh, right, actually that changes how we need this to work then.  Basically the callback will stop when this lambda starts going *up* in value.  So, the expression we need here is a little different---we need something that's large (and falling, potentially) when we are far away from 10*the unit ball, and once we get inside it needs to go back up.  We can do that like this:

```
  double minVal = arma::abs(coordinates).min();
  return (minVal < 10.0) ? DBL_MAX : minVal;
```

I think that that will work, want to give it a try?

Reply:
Agreed